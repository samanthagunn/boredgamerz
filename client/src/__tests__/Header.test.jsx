import React, { useState as useStateMock } from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { useAuth0 } from "@auth0/auth0-react";
import renderer from "react-test-renderer";
import Header from "../components/header";
jest.mock("@auth0/auth0-react");
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: jest.fn(),
}));

const user = {
  name: "Christian Glassiognon",
  given_name: "Christian",
  picture:
    "https://lh3.googleusercontent.com/a-/AOh14GiG4KuyPH8-ToSqzoBLwPTvb-px4DboWJn_Gujs=s96-c",
};
const setState = jest.fn();
describe("The header component...", () => {
  describe("that is user authenticated...", () => {
    beforeAll(() => {
      useStateMock.mockImplementation((init) => [init, setState]);
      useAuth0.mockReturnValue({
        logout: jest.fn(),
        loginWithRedirect: jest.fn(),
        isAuthenticated: true,
        isLoading: false,
        user: user,
      });
    });
    test("will render without crashing.", () => {
      const { baseElement } = render(<Header />);
      expect(baseElement).toBeDefined();
    });
    test("will display correct header.", () => {
      const tree = renderer.create(<Header />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    test("will call state for popover.", () => {
      const { getByText } = render( <Header />);
      fireEvent.click(getByText("Profile"));
      expect(setState).toHaveBeenCalledTimes(1);
    });
  });
  describe("that is not user authenticated...", () => {
    beforeAll(() => {
      useAuth0.mockReturnValue({
        logout: jest.fn(),
        loginWithRedirect: jest.fn(),
        isAuthenticated: false,
        isLoading: false,
        user: user,
      });
    });
    test("will render without crashing.", () => {
      const { baseElement } = render(<Header />);
      expect(baseElement).toBeDefined();
    });
    test("will display correct header.", () => {
      const tree = renderer.create(<Header />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    test("will call login when pressed", () => {
      const { getByText } = render(<Header />);
      fireEvent.click(getByText("Login"));
      expect(useAuth0().loginWithRedirect).toHaveBeenCalled();
    });
  });
});
