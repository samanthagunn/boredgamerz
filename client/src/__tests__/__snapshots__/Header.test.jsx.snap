// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The header component... that is not user authenticated... will display correct header. 1`] = `
<ion-toolbar
  color="primary"
>
  <a
    href="/home"
  >
    <img
      alt="BoredGamerz Logo"
      className="logo-hero"
      src="https://storage.googleapis.com/boredgamerz_assets/BoredGamerzLogo_72-2.png"
    />
  </a>
  <ion-buttons
    slot="primary"
  >
    <ion-button
      onClick={[Function]}
    >
      Login
    </ion-button>
  </ion-buttons>
</ion-toolbar>
`;

exports[`The header component... that is user authenticated... will display correct header. 1`] = `
<ion-toolbar
  color="primary"
>
  <a
    href="/home"
  >
    <img
      alt="BoredGamerz Logo"
      className="logo-hero"
      src="https://storage.googleapis.com/boredgamerz_assets/BoredGamerzLogo_72-2.png"
    />
  </a>
  <ion-buttons
    slot="primary"
  >
    <ion-button
      href="/profile/games"
      onClick={[Function]}
    >
      My Games
    </ion-button>
    <ion-button
      href="/games"
      onClick={[Function]}
    >
      Find Games
    </ion-button>
    <ion-button
      href="/games/create"
      onClick={[Function]}
    >
      Create A Game
    </ion-button>
    <ion-button
      onClick={[Function]}
    >
      Profile
    </ion-button>
    <img
      alt="profile"
      className="user-image"
      height="50px"
      onClick={[Function]}
      src="https://lh3.googleusercontent.com/a-/AOh14GiG4KuyPH8-ToSqzoBLwPTvb-px4DboWJn_Gujs=s96-c"
    />
  </ion-buttons>
</ion-toolbar>
`;
